android/filament-utils-android/src/main/cpp/Utils.cpp

```diff
@@ -96,3 +72,24 @@ Java_com_google_android_filament_utils_KtxLoader_nCreateSkybox(JNIEnv* env, jcla
     }, bundle);
     return (jlong) Skybox::Builder().environment(cubemap).showSun(true).build(*engine);
 }
+
+
+JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void*) {
+    JNIEnv* env;
+    if (vm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6) != JNI_OK) {
+        return -1;
+    }
+
+    jclass c = env->FindClass("com/google/android/filament/utils/KtxLoader");
+    if (c == nullptr) return JNI_ERR;
+
+    static const JNINativeMethod methods[] = {
+        {"nCreateTexture", "(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateTexture)},
+        {"nCreateIndirectLight", "(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateIndirectLight)},
+        {"nCreateSkybox", "(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateSkybox)},
```

Comment:
this makes me nervous, how do we know for sure we don't have typo in one of the signature? Wouldn't that be detected only at runtime? Or will it be detected during the RegisterNatives call?

Reply:
Actually that's the nice thing about `RegisterNatives`, it detects signature mismatches immediately.

Reply:
FWIW all the JNI of Android is written this way ;-)